File: MicroServices/MicroServices/referenciel-service/src/main/java/com/bfi/referentiel/service/EmployeeService.java
package com.bfi.referentiel.service;

import com.bfi.referentiel.common.dto.EmployeeDTO;

import java.util.List;

public interface EmployeeService {

    void addEmployee(EmployeeDTO employee);

    void updateEmployee(EmployeeDTO employee);

    void deleteEmployee(Long id);

    List<EmployeeDTO> getAllEmployee();

    List<EmployeeDTO> findEmployeesByAgencyId(Long id);
}

File: CartagoFrontV2-master/src/app/services/referenciel/employee.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { Employee } from '../../models/employee';

@Injectable({
  providedIn: 'root',
})
export class EmployeeService {
  constructor(private http: HttpClient) {}

  getAllEmployees(): Observable<Employee[]> {
    return this.http.get<Employee[]>('http://localhost:8080/Employees');
  }

  addEmployee(employee: Employee): Observable<Employee> {
    return this.http.post<Employee>(
      'http://localhost:8080/Employees/',
      employee
    );
  }

  updateEmployee(employee: Employee): Observable<Employee> {
    return this.http.put<Employee>(
      'http://localhost:8080/Employees/',
      employee
    );
  }

  deleteEmployee(id: number): Observable<Employee> {
    return this.http.delete<Employee>('http://localhost:8080/Employees/' + id);
  }
}
